src/api.py

```diff
@@ -1,26 +1,96 @@
-from flask import Flask, request
+import os
+
+from celery.result import AsyncResult
+from fastapi import FastAPI
+from fastapi.params import Query
+from pydantic import BaseModel
+from starlette.responses import HTMLResponse
+
 from src import service
```

Comment:
IMO it would be nicer if the python package would be named `tmt_web` or `tmt.web` (this one rather not, it could potentially collide with code in https://github.com/teemtee/tmt/). So the imports would look like e.g. `from tmt_web import service`.

Reply:
I don't think this is package name, but rather relative import from path, no? Should we add project metadata?

Reply:
Yeah, you're right, it's a relative import, +1 for project metadata.

Reply:
Relative imports must begin with a dot (eg. "from .src import service"), so this is a standard module/package import call. Python does add the current working directory to its list of module import lookup paths with the highest priority, so in the end it behaves the same way.

It's still a bad practice though and i'd suggest rearranging the project into what's the de-facto standard python project structure nowadays:

```
pyproject.toml
README.md
src/
  tmt_web/
    __init__.py
    api.py
    generators/
    ...
```

where a minimal working `pyproject.toml` could look something like this:
```
[build-system]
requires = ["setuptools>=64"]
build-backend = "setuptools.build_meta"

[project]
name = "tmt_web"
description = "TMT web app"
readme = "README.md"
requires-python = ">=3.10"
version = "0.0.1"
dependencies = [
    "tmt~=1.35",
    "fastapi~=0.112",
    "httpx~=0.27",
    "uvicorn~=0.30",
    "celery[redis]~=5.4",
]

[tool.setuptools.packages.find]
where = ["src"]
```

Reply:
Thanks, I'm already working on it. +1 for standard src/tmt_web/

Reply:
btw are there any conventions about underscore vs dash vs .. ?

Reply:
project names should prefer dashes, while package/module/file names must use underscores

Reply:
Note that in PyPI the names are normalized, e.g. `jaraco.context` can be accessed as:
- https://pypi.org/project/jaraco.context/
- https://pypi.org/project/jaraco-context/
- https://pypi.org/project/JaRaCo_CoNtExT/
- etc.

and thus similarly there is no effect of what you choose in the `dependencies` as well. What is important though is making sure the `sdist` package is normalized to `_`, which I think only started happening in the recent  1-2 years. I am not sure about `setuptools`, but definitely going for `hatchling` build system does that for us.

But indeed usual convention for `project.name` is to use dashes `-`.

Reply:
@tkoscieln I kinda forgot about this. Let's change use a dash :)